libs/barista-components/experimental/datepicker/src/timeinput.spec.ts

```diff
@@ -0,0 +1,291 @@
+import {
+  createComponent,
+  dispatchFakeEvent,
+  dispatchKeyboardEvent,
+  typeInElement,
+} from '@dynatrace/testing/browser';
+
+describe('DtTimeInput', () => {
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [DtDatepickerModule],
+        declarations: [SimpleTimeInputTestApp],
+      });
+
+      TestBed.compileComponents();
+    }),
+  );
+
+  describe('basic behavior', () => {
+    let fixture: ComponentFixture<SimpleTimeInputTestApp>;
+    let component: SimpleTimeInputTestApp;
+    let hourEl: HTMLInputElement;
+    let minuteEl: HTMLInputElement;
+
+    beforeEach(() => {
+      fixture = createComponent(SimpleTimeInputTestApp);
+      component = fixture.componentInstance;
+      hourEl = component.timeInput._hourInput.nativeElement;
+      minuteEl = component.timeInput._minuteInput.nativeElement;
+      fixture.detectChanges();
+    });
+
+    describe('disabled input property', () => {
+      it('should not be usable when disabled', fakeAsync(() => {
+        component.disabled = true;
+        fixture.detectChanges();
+        tick();
+        expect(hourEl.disabled).toBeTruthy();
+        expect(hourEl.getAttribute('aria-disabled')).toBeTruthy();
+        expect(minuteEl.disabled).toBeTruthy();
+        expect(minuteEl.getAttribute('aria-disabled')).toBeTruthy();
+      }));
+    });
+
+    describe('timeChange event', () => {
+      it('should emit a timechange event when the hour and minute inputs are changed', fakeAsync(() => {
+        const changeSpy = jest.fn();
+        component.timeInput.timeChange.subscribe(changeSpy);
+        fixture.detectChanges();
+        expect(changeSpy).not.toHaveBeenCalled();
+        component.timeInput.hour = 23;
+        component.timeInput.minute = 55;
+        fixture.detectChanges();
+        dispatchFakeEvent(hourEl, 'blur');
+        expect(changeSpy).toHaveBeenCalledTimes(1);
+      }));
+
+      it('should emit a timechange event when the hour and minute inputs are reset to empty', fakeAsync(() => {
+        const changeSpy = jest.fn();
+        component.timeInput.timeChange.subscribe(changeSpy);
+        fixture.detectChanges();
+        expect(changeSpy).not.toHaveBeenCalled();
+        component.timeInput.hour = null;
+        component.timeInput.minute = null;
+        fixture.detectChanges();
+        dispatchFakeEvent(minuteEl, 'blur');
+        expect(changeSpy).toHaveBeenCalledTimes(1);
+      }));
+    });
+
+    describe('Focus switch', () => {
+      it('should switch focus from the hour to the minute input when typing in 2 digits in the hour input', fakeAsync(() => {
+        component.timeInput.minute = null;
+        component.timeInput.hour = 14;
+        fixture.detectChanges();
+
+        dispatchKeyboardEvent(hourEl, 'keyup', NUMPAD_ONE);
+        fixture.detectChanges();
+        tick();
+
+        expect(document.activeElement).toBe(minuteEl);
+      }));
```

Comment:
It actually did not matter which key is used in there, but I needed it in order to trigger the 'keyup' event from the input (the focus switch is handled on keyup), since changing the model value programmatically wouldn't do so.